packages/tree-widget/src/components/CategoriesTree.tsx

```diff
@@ -0,0 +1,91 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import * as React from "react";
+import {
+  IModelApp,
+  IModelConnection,
+  Viewport,
+} from "@bentley/imodeljs-frontend";
+import { CategoryTree, toggleAllCategories } from "@bentley/ui-framework";
+import { IconButton } from "./IconButton";
+import { SearchBar } from "./search-bar/SearchBar";
+import { useTreeFilteringState } from "./TreeFilteringState";
+import "./CategoriesTree.scss";
+import { TreeWidget } from "../TreeWidget";
+
+interface CategoriesTreeComponentProps {
+  iModel: IModelConnection;
+  allViewports?: boolean;
+  activeView?: Viewport;
+  enablePreloading?: boolean;
+}
+
+export function CategoriesTreeComponent(props: CategoriesTreeComponentProps) {
+  const {
+    searchOptions,
+    filterString,
+    activeMatchIndex,
+    onFilterApplied,
+    filteredProvider,
+  } = useTreeFilteringState();
+
+  const showAll = React.useCallback(async () => {
+    return toggleAllCategories(
+      IModelApp.viewManager,
+      props.iModel,
+      true,
+      undefined,
+      true,
+      filteredProvider
+    );
+  }, [props.iModel, filteredProvider]);
+
+  const hideAll = React.useCallback(async () => {
+    return toggleAllCategories(
+      IModelApp.viewManager,
+      props.iModel,
+      false,
+      undefined,
+      true,
+      filteredProvider
+    );
+  }, [props.iModel, filteredProvider]);
+
+  return (
+    <>
+      <SearchBar
+        value=""
+        className="ui-test-app-search-bar"
+        valueChangedDelay={500}
+        placeholder={TreeWidget.translate("search")}
+        title={TreeWidget.translate("searchForSomething")}
+        filteringInProgress={searchOptions.isFiltering}
+        onFilterCancel={searchOptions.onFilterCancel}
+        onFilterClear={searchOptions.onFilterCancel}
+        onFilterStart={searchOptions.onFilterStart}
+        onSelectedChanged={searchOptions.onResultSelectedChanged}
+        resultCount={searchOptions.matchedResultCount ?? 0}
+      >
```

Comment:
Just an fyi... when (if) I get some free time, I will pull out the SearchBar, and possibly move it to core.

Reply:
Sounds good.